### ISSUE TITLE
check mark of checkedtextview gets reset in MultiCheck after scrolling

### ISSUE THREAD

**Issue by GiorgiSheklashvili** [thoughtbot/expandable-recycler-view#72]
In MultiCheck when there is enough checkedtextviews to be scrolled, check mark changes position or gets reseted at all. This happens only to the items that are not visible.
Is there any fix to this?

**Comment by Aksi0m** [comment]
@giusha did you ever find a solution to your problem?

**Comment by giusha** [comment]
@Aksi0m yes, in my case it was somehow related to the fact that I was using stringeSets to send list from another fragment, and in adapter I was converting it to List<string>. Because sharedpreferences can keep stringSets. But then I was forced to replace it with simple List<String> and this problem disappeared. So in other words I changed from sharedpreferences.putstringset  to saving and restoring List<string> with JSON. So don't use stringset if it's the case.

**Comment by Aksi0m** [comment]
I dont use String sets. I am not sure that caused the problem you had, sounds very strange. What adapter from the lib do you use? Would it be a problem if you could share the code, just the adapter?

**Comment by giusha** [comment]
I'm using MultiCheckExpandableGroup.
https://codepaste.net/m8pyhp
This is my adapter, In my constructor I'm trying to sum up the number of same Strings and write it in textview.

**Comment by Aksi0m** [comment]
@giusha you dont use any of the adapters provided by this library. Your adapter is the default RecycleView adapter. Maybe thats because your solution actually works, because I cant figure out why my is not working, I use `CheckableChildRecyclerViewAdapter` and it is not working.

**Comment by giusha** [comment]
I had problem in that adapter too. This is checkableChildRecyclerViewAdapter https://codepaste.net/4aqvu5 and it's working fine . https://codepaste.net/7i4kjw
https://codepaste.net/47z2xq  . I had problem in this adapter too and after changing to List<string> every adapter problem disappeared somehow, including this one. 
final List<Category> categories = presenter.getCategories();
        categoryAdapter = new CategoryAdapter(categories);
This is how I'm creating adapter. 
this is category class: https://codepaste.net/roang4
and this is submenu class: https://codepaste.net/xigrsb
Hope it helps

**Comment by Aksi0m** [comment]
@giusha thank you for your time and help. I jut succeeded to find my problem. My `CheckedTextView` didnt use `android:checkMark` attribute, instead it used a drawable `android:drawableLeft` and that is why it didnt update correctly in the list and was also the cause of the random check states while scrolling. The goal was to have the checkbox on the left and this did not work with the current implementation, instead I created a custom` CheckedTextView` where in the implementation I changed the gravity of the `CheckBox`. Now everything works smoothly.
